zipline/utils/pandas_utils.py

```diff
@@ -222,3 +223,45 @@ def clear_dataframe_indexer_caches(df):
             delattr(df, attr)
         except AttributeError:
             pass
+
+
+def categorical_df_concat(df_list, inplace=False):
+    """
+    Prepare list of pandas DataFrames to be used as input to pd.concat.
+    Ensure any columns of type 'category' have the same categories across each
+    dataframe.
+
+    Parameters
+    ----------
+    df_list : list
+        List of dataframes with same columns.
+    inplace : bool
+        True if input list can be modified. Default is False.
+
+    Returns
+    -------
+    concatenated : df
+        Dataframe of concatenated list.
+    """
+
+    if not inplace:
+        df_list = deepcopy(df_list)
+
+    # Assert each dataframe has the same columns/dtypes
+    df = df_list[0]
+    if not all([(df.dtypes.equals(df_i.dtypes)) for df_i in df_list[1:]]):
+        raise ValueError("Input DataFrames must have the same columns/dtypes.")
+
+    categorical_columns = df.columns[df.dtypes == 'category']
+
+    for col in categorical_columns:
+        new_categories = sorted(
+            set().union(
+                *(frame[col].cat.categories for frame in df_list)
+            ) - {None}
```

Comment:
Right.  This is a known issue that we've punted on updating because it's a breaking change for consumers of the API.  In `LabelArray.as_categorical`, for example,  we [explicitly silence this warning](https://github.com/quantopian/zipline/blob/master/zipline/lib/labelarray.py#L324).

The issue here is that when we first added support for `Classifiers` in Pipeline, the api allowed for consumers to specify how missing data should be represented.  For strings data, we default to `None` as the missing value, which means that pipeline results for string-dtype classifiers contain `None` for locations where we don't have any data.  When we upgraded to pandas 18, pandas started emitting warnings about `None` values in categories, but we ignored the warning because what pandas wants us to do instead is let it use its preferred value of `NaN`, which is a significant behavioral change for users.  I think we want to revisit that issue at some point, but I don't think right now is the time for that change (my guess is that we'll deal with this the next time we upgrade pandas).

For the time being, my recommendation would be to wrap this for-loop in the same `ignore_pandas_categorical_nan_warning` block that we use in LabelArray.